jwst/extract_2d/grisms.py

```diff
@@ -167,52 +170,66 @@ def extract_tso_object(input_model,
             extract_y_min = 0
             extract_y_max = extract_height - 1
         else:
-            extract_y_min = extract_y_center - 34  # always return source at pixel 34 in cutout.
+            extract_y_min = extract_y_center - 34  # always return source at row 34 in cutout
             extract_y_max = extract_y_center + extract_height - 34 - 1
 
+        # Check for bad results
         if (extract_y_min > extract_y_max):
             raise ValueError("Something bad happened calculating extraction y-size")
 
-        # limit the bounding box to the detector
+        # Limit the bounding box to the detector edges
         ymin, ymax = (max(extract_y_min, 0), min(extract_y_max, input_model.meta.subarray.ysize))
         xmin, xmax = (max(xmin, 0), min(xmax, input_model.meta.subarray.xsize))
-        log.info("xmin, xmax: {} {}  ymin, ymax: {} {}".format(xmin, xmax, ymin, ymax))
 
-        # the order and source position are put directly into
-        # the new wcs for the subarray for the forward transform
+        # The order and source position are put directly into the new WCS of the subarray
+        # for the forward transform.
+        #
+        # NOTE NOTE NOTE  2020-02-14
+        # We would normally use x-axis (along dispersion) extraction limits calculated
+        # above based on the min/max wavelength range and the source position to do the
+        # subarray extraction and set the subarray WCS accordingly. HOWEVER, the NIRCam
+        # team has asked for all data along the dispersion direction to be included in
+        # subarray cutout, so here we override the xmin/xmax values calculated above and
+        # instead hardwire the extraction limits for the x (dispersion) direction to
+        # cover the entire range of the data and use this new minimum x value in the
+        # subarray WCS transform. If the team ever decides to change the extraction limits,
+        # the following two constants must be modified accordingly.
+        xmin_ext = 0  # hardwire min x for extraction to zero
+        xmax_ext = input_model.data.shape[-1]  # hardwire max x for extraction to size of data
```

Comment:
It's a little confusing that `xmax_ext` here is a slice limit, rather than the actual upper limit.  That results in inconsistent notation when extracting a slice, e.g. `ymin: ymax + 1, xmin_ext:xmax_ext`.  If you subtract 1 here, you would need to add 1 when taking the slice; that would be more verbose, but perhaps less confusing because the notation for both x and y would be consistent.

Reply:
I thought about that too while writing the code. I left it this way for simplicity, but I think I will go back and update it so that it works consistently as slice indexes.

Reply:
Aren't the bounding box limits inclusive?  I thought that was why `ymax` was the actual upper limit.

Reply:
Yes, bounding box limits are inclusive, so in the y-axis they go from -0.5 to 63.5 (in 0-indexed python pixels) for the data array that is 64 pixels high. In the slicing done below, the upper limit on y is ymax+1, which causes the last pixel in the cutout to be number 63 (0-indexed). This change to setting "Shift(xmin_ext)" has no affect on the bounding box. It's still -0.5 < x < 2047.5 and -0.5 < y < 63.5, as it was before.